pkg/diagnostics/collectors.go

```diff
@@ -435,9 +437,98 @@ func (c *collectorFactory) pingHostCollector(hostIP string) *Collect {
 					Command: []string{"/bin/sh", "-c"},
 					Args:    argsPing,
 				}},
+			}, Timeout: "20s",
+		},
+	}
+}
+
+// vmsAccessCollector will connect to API server first, then collect vsphere-cloud-controller-manager logs
+// on control plane node
+func (c *collectorFactory) vmsAccessCollector(controlPlaneConfiguration v1alpha1.ControlPlaneConfiguration) *Collect {
+	controlPlaneEndpointHost := controlPlaneConfiguration.Endpoint.Host
+	taints := controlPlaneConfiguration.Taints
+	tolerations := makeTolerations(taints)
+
+	makeConnection := fmt.Sprintf("curl --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" https://%s:6443/", controlPlaneEndpointHost)
+	getLogs := "kubectl logs -n kube-system -l k8s-app=vsphere-cloud-controller-manager"
+	args := []string{fmt.Sprintf("%s && %s", makeConnection, getLogs)}
+	return &Collect{
+		RunPod: &runPod{
+			Name:      "check-cloud-controller",
+			Namespace: constants.EksaDiagnosticsNamespace,
+			PodSpec: &v1.PodSpec{
+				Containers: []v1.Container{{
+					Name:    "check-cloud-controller",
+					Image:   c.DiagnosticCollectorImage,
+					Command: []string{"/bin/sh", "-c"},
+					Args:    args,
+				}},
+				ServiceAccountName: "default",
+				HostNetwork:        true,
+				Tolerations:        tolerations,
+				Affinity: &v1.Affinity{
+					NodeAffinity: &v1.NodeAffinity{
+						PreferredDuringSchedulingIgnoredDuringExecution: []v1.PreferredSchedulingTerm{
+							{
+								Weight: 10,
+								Preference: v1.NodeSelectorTerm{
+									MatchExpressions: []v1.NodeSelectorRequirement{{
+										Key:      "node-role.kubernetes.io/control-plane",
```

Comment:
curious, why does this collector need to run in the cp?

Reply:
To deal with the problem that vms have no access to api server, the analyzer need info from vsphere-cloud-controller-manager.log on cp, which should be collected by a log collecter, ideally. However, that log collector must collect vsphere-cloud-controller-manager.log from control plane and worker nodes, then generate corresponding  amount of log files with almost the same title. The analyzer is not able to figure out which log file it should work with. Therefore, we deploy a pod and run it in the cp to get the exact log that the analyzer needs to analyze for this issue.

Reply:
What do you mean exactly by `vsphere-cloud-controller-manager.log`? Are you reading the logs from the node? I thought you were using `kubectl logs` to retrieve the logs from the cloud manager pod.

Reply:
> What do you mean exactly by `vsphere-cloud-controller-manager.log`? Are you reading the logs from the node? I thought you were using `kubectl logs` to retrieve the logs from the cloud manager pod.

Sorry I didn't make it clear. Once the issue that VMs have no access to api server happens, only log that contains the corresponding error message is the vsphere-cloud-controller-manager.log from the Control Plane Node. Ideally, log collector should read logs from the nodes by using `kubectl logs` and it's doing this job as well now. Then analyzer will open the log files and catch the error message. However, the log files we got not only from control plane but also from worker nodes. The analyzer cannot figure out which log file it should analyze so it does its job on every log files, and gives us multiple results, which is confused and unexpected. Therefore, we deploy a pod and run it in the cp to directly get the vsphere-cloud-controller-manager.log from Control Plane Node. The analyzer can analyze for this issue and give us accurate diagnostic result.

Reply:
Then I'm missing something. Isn't this pod collector just running `kubectl logs`?

Reply:
> Then I'm missing something. Isn't this pod collector just running `kubectl logs`?

You're good. I should clarify it first. Yes and No to answer your question. Yes, the collector run "kubectl logs" to obtain logs. No, becasue it's not just run "kubectl logs", it also need to connect control plane first.